Source/CarthageKit/Project.swift

```diff
@@ -117,6 +118,10 @@ public enum ProjectEvent {
 	/// Building the project is being skipped, since the project is not sharing
 	/// any framework schemes.
 	case skippedBuilding(ProjectIdentifier, String)
+
+	/// Installing of a binary framework is being skipped because of an inability
+	/// to verify that it was built with a compatible Swift version.
+	case skippedInstallingBinaries(ProjectIdentifier, String)
```

Comment:
I don't think they even need a case on `CarthageError`. If you create the internal error, you can pass that to the `ProjectEvent` case.